source/concurrent/SynchronizedQueue.ooc

```diff
@@ -12,8 +12,18 @@ import threading/[Thread, Mutex, WaitCondition]
 SynchronizedQueue: class <T> extends Queue<T> {
 	_mutex := Mutex new()
 	_backend: Queue<T>
-	count ::= this _backend count
-	empty ::= this _backend empty
+	count: Int { get {
+		this _mutex lock()
```

Comment:
Why locks here? As soon as you unlock the mutex you can't be sure anyway if `count` is still valid.